### PR TITLE
Component jobs: use new templates

### PR DESCRIPTION
**Pull request by Halamix2**
**Description**

Changes proposed in this pull request:

- rework generic component jobs to use new templating

**Related issue(s)**

#3452

### CODE REVIEW

File: development/tools/pkg/rendertemplates/genericcomponent.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+				// check if the jobConfig is in expected for component jobs format
+				if job.JobConfig["name"] == nil && job.JobConfig["path"] != nil {
+					hasComponentJobs = true
+					// generate component jobs
+
+					repository := strings.Split(repo.RepoName, "/")[2]
+					nameSuffix := repository + "-" + strings.Replace(job.JobConfig["path"].(string), "/", "-", -1)
+
+					// generate pre- and post-submit jobs for the next release
+					if ReleaseMatches(global["nextRelease"], job.jobConfigPre["release_since"], job.jobConfigPre["release_until"]) {
```
Comment (dekiel) [inline development/tools/pkg/rendertemplates/genericcomponent.go:33]: Can we search for release_since and release_until in whole JobConfig not only in jobConfigPre and jobConfigPost? If you want pre and post generate for same set of release which I think is our case, you can set the boundaries in jobConfig to apply for both instead setting them twice for pre and post.

**Review by dekiel — CHANGES_REQUESTED** [review]

File: development/tools/cmd/rendertemplates/main.go [hunk 2]
```diff
@@ -326,16 +190,29 @@ func hasProwjobType(r []Repo, prowjobtype string) bool {
...
+func getRunID(name interface{}) string {
+	jobName := name.(string)
+	if len(jobName) > 63 {
+		jobName = jobName[0:63]
+		for jobName[len(jobName)-1:] == "-" {
+			jobName = jobName[:len(jobName)-1]
+		}
+	}
+	return "\"" + jobName + "\""
+	//{{ $runIDtrunc := trunc 63 .JobConfig.name}}{{ if hasSuffix "-" $runIDtrunc}}{{ substr 0 (sub ( (len $runIDtrunc) | int ) 1) $runIDtrunc | quote }}{{else}}{{ $runIDtrunc | quote}}{{end}}
```
[the remark this answers is not in the document]
Comment (Halamix2) [reply development/tools/cmd/rendertemplates/main.go]: Done

File: development/tools/pkg/rendertemplates/genericcomponent.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
+package rendertemplates
+
+import (
+	"strings"
+)
+
+// changeExtraRefsBase changes base_ref to base string for each extra_ref
+func (j *Job) changeExtraRefsBase(base string) {
+	if j.JobConfig["extra_refs"] != nil {
+		for extraRefIndex := range j.JobConfig["extra_refs"].(map[interface{}]interface{}) {
```
[the remark this answers is not in the document]
Comment (Halamix2) [reply development/tools/pkg/rendertemplates/genericcomponent.go:10]: Linter won't let me use `extraRefIndex, _`

**Review by dekiel — APPROVED** [review]

**Pull request merged**